scripts/version.js

```diff
@@ -1,16 +1,64 @@
-// core packages
+const { execSync } = require('child_process');
+const fs = require('fs');
 const path = require('path');
-const util = require('util');
-const exec = util.promisify(require('child_process').exec);
+const semver = require('semver');
 
-const packageJson = require('./../package.json');
+const PACKAGE_JSON_FILE = path.resolve(__dirname, '..', 'package.json');
+const PACKAGE_LOCK_FILE = path.resolve(__dirname, '..', 'package-lock.json');
 
-async function main() {
-  const { stdout, stderr } = await exec(
-    `npm i -E simple-icons@${packageJson.version}`
-  );
+function stringifyJson(object) {
+  return JSON.stringify(object, null, 2);
+}
+
+function updateSimpleIconsDependency() {
+  try {
+    execSync('npm uninstall simple-icons');
+    execSync('npm install --save-dev --save-exact simple-icons');
+  } catch (err) {
+    return err;
+  }
+}
+
+function updateVersionNumber() {
+  try {
+    const packageFileRaw = fs.readFileSync(PACKAGE_JSON_FILE).toString();
+    const packageLockRaw = fs.readFileSync(PACKAGE_LOCK_FILE).toString();
+
+    const packageFile = JSON.parse(packageFileRaw);
+    const packageLock = JSON.parse(packageLockRaw);
+
+    const currentVersion = packageFile.version;
+    const simpleIconsVersion = packageFile.devDependencies['simple-icons'];
+    if (semver.gt(simpleIconsVersion, currentVersion)) {
+      packageFile.version = simpleIconsVersion;
+      packageLock.version = simpleIconsVersion;
+    } else {
+      const err = new Error(`Version already exists (${currentVersion})`);
+      return [null, err];
+    }
+
+    fs.writeFileSync(PACKAGE_JSON_FILE, stringifyJson(packageFile) + '\n');
+    fs.writeFileSync(PACKAGE_LOCK_FILE, stringifyJson(packageLock) + '\n');
+    return [simpleIconsVersion, null];
+  } catch (err) {
+    return [null, err];
+  }
+}
+
+function main() {
+  const errorA = updateSimpleIconsDependency();
+  if (errorA) {
+    console.error('Failed to update simple-icons to latest version:', errorA);
+    process.exit(1);
+  }
+
+  const [newVersion, errorB] = updateVersionNumber();
+  if (errorB) {
+    console.error('Failed to version bump this package:', errorB);
+    process.exit(1);
+  }
 
-  if (stderr) console.error(`error: ${stderr}`);
+  console.log(`::set-output name=NEW_VERSION::${newVersion}`);
```

Comment:
Why not output only the version, catching it in the workflow and moving this Github Actions-dependent log to the workflow?

Reply:
Seems complicated, I'm not against it if it can be done in 1 or 2 lines though but I'm not sure how

Reply:
Would be:

```suggestion
  console.log(newVersion);
```

and in the workflow:

```bash
newVersion="$(node scripts/version.js)"
echo ::set-output name=NEW_VERSION::$newVersion
```

Reply:
That is a lot easier then I was thinking :sweat_smile: Thanks for the suggestion, implemented this in 3d293890dab012763a788620da1a7e2774923ae8 :+1: